loki/bulk/item.py

```diff
@@ -113,6 +113,13 @@ def __eq__(self, other):
     def __hash__(self):
         return hash(self.name)
 
+    def clear_cached_property(self, property_name):
```

Comment:
The fact that we need this is worrying. If a property needs to be wiped, it might not need to be cached in the first place, no? From what I can see this is only the `Imports` in a module/routine spec, right? Could we simply live with re-deriving these on the fly, as they are accessed quite rarely, I would think?

Reply:
I suspect we could - the reason I didn't simply remove the caching is that we do in fact depend on the caching behaviour, at least after applying dependency injection (and the associated renaming). So, a callback mechanism that updates items if they are renamed might be something for the Scheduler wishlist. See LOKI-42 where this was flagged a while ago. Might be worth checking the runtime impact as well.

Reply:
Ok, that sounds sensible. We'll address this in a follow-up.